common/djangoapps/embargo/api.py

```diff
@@ -0,0 +1,119 @@
+"""
+The Python API layer of the country access settings. Essentially the middle tier of the project, responsible for all
+business logic that is not directly tied to the data itself.
+
+This API is exposed via the middleware(emabargo/middileware.py) layer but may be used directly in-process.
+
+"""
+import logging
+import pygeoip
+
+from django.core.cache import cache
+from django.conf import settings
+from embargo.models import CountryAccessRule, IPFilter, RestrictedCourse
+
+log = logging.getLogger(__name__)
+
+
+def _check_ip_lists(ip_addr):
+    """
+    Check whether the user is embargoed based on the IP address.
+
+    Args:
+        ip_addr (str): The IP address the request originated from.
+
+    Returns:
+        A True if the user is not blacklisted, otherwise False
+
+    """
+    # If blacklisted, immediately fail
+    if ip_addr in IPFilter.current().blacklist_ips:
```

Comment:
@wedaly this is the part I was concerned with regarding black lists and white lists. I see the logic in the Country model object around white and black lists, I get the impression they're not intended to be used together? Maybe something isn't implemented just yet?